File: rules/git/commit-message-describes-diff/fixtures/flag-plus-strict-off/before/src/cli/report.ts
import { collectReport } from "../report/collect.ts";
import type { Report } from "../report/types.ts";

function renderText(report: Report): string {
  const lines = [`${report.project}: ${report.findings.length} finding(s)`];
  for (const f of report.findings) lines.push(`  ${f.file}:${f.line}  ${f.rule}  ${f.message}`);
  return lines.join("\n");
}

export async function reportCommand(args: string[]): Promise<number> {
  const root = args.find((a) => !a.startsWith("--")) ?? process.cwd();
  const report = await collectReport(root);
  process.stdout.write(`${renderText(report)}\n`);
  return report.findings.length === 0 ? 0 : 1;
}

File: rules/git/commit-message-describes-diff/fixtures/version-bump-alongside/before/src/report.rs
use clap::Args;
use serde::Serialize;

use crate::collect::{collect, Finding};

#[derive(Args)]
pub struct ReportArgs {
    /// Directory to report on.
    #[arg(default_value = ".")]
    pub root: String,
}

#[derive(Serialize)]
pub struct Report {
    pub project: String,
    pub findings: Vec<Finding>,
}

fn render_text(report: &Report) -> String {
    let mut lines = vec![format!("{}: {} finding(s)", report.project, report.findings.len())];
    for f in &report.findings {
        lines.push(format!("  {}:{}  {}  {}", f.file, f.line, f.rule, f.message));
    }
    lines.join("\n")
}

pub fn run(args: &ReportArgs) -> anyhow::Result<i32> {
    let report = collect(&args.root)?;
    println!("{}", render_text(&report));
    Ok(if report.findings.is_empty() { 0 } else { 1 })
}
